webapi/CopilotChat/Skills/TokenUtilities.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Azure.AI.OpenAI;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Connectors.AI.OpenAI.Tokenizers;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace SemanticKernel.Service.CopilotChat.Skills;
+
+/// <summary>
+/// Utility methods for token management.
+/// </summary>
+public static class TokenUtilities
+{
+    /// <summary>
+    /// Semantic dependencies of ChatSkill.
+    ///  If you add a new semantic dependency, please add it here.
+    /// </summary>
+    public static readonly Dictionary<string, string> semanticFunctions = new()
+    {
+        // TODO: [Issue #2106] Calculate token usage for planner dependencies.
+        { "SystemAudienceExtraction", "audienceExtraction" },
+        { "SystemIntentExtraction", "userIntentExtraction" },
+        { "SystemMetaPrompt", "metaPromptTemplate" },
+        { "SystemCognitive_WorkingMemory", "workingMemoryExtraction" },
+        { "SystemCognitive_LongTermMemory", "longTermMemoryExtraction" },
+        { "SystemCompletion", "responseCompletion"}
+    };
+
+    /// <summary>
+    /// Gets dictorary containing empty token usage totals.
+    /// Use for responses that are hardcoded and/or do not have semantic (token) dependencies.
+    /// </summary>
+    internal static Dictionary<string, int> EmptyTokenUsages()
+    {
+        var emptyTokenUsages = new Dictionary<string, int>();
```

Comment:
semanticFunctions.Values.ToDictionary(v => v, v => 0);

Also please verify whether you intend to dictionary key comparison be case invariant or not. (currently not).

Reply:
oh awesome, didn't know about this syntax `.Values.ToDictionary(v => v, v => 0);`. Thanks! Added comparison definition too.